src/lightcurvelynx/simulate.py

```diff
@@ -221,20 +354,157 @@ def simulate_lightcurves(
+    futures_or_results = executor.map(_simulate_lightcurves_batch, batches)
+    result_list = []
+    for res in futures_or_results:
+        if hasattr(res, "result") and callable(res.result):  # A Future
+            result_list.append(res.result())
+        else:  # A direct result
+            result_list.append(res)
+    return result_list
+
+
+def simulate_lightcurves(
+    model,
+    num_samples,
+    obstable,
+    passbands,
+    *,
+    obstable_save_cols=None,
+    param_cols=None,
+    apply_obs_mask=False,
+    time_window_offset=None,
+    output_file_path=None,
+    rng=None,
+    executor=None,
+    num_jobs=None,
+    batch_size=1_000,
+):
+    """Generate a number of simulations of the given model and information
+    from one or more surveys. The result data can either be returned directly
+    (as a single nested data frame) or saved to file(s).
+
+    Parameters
+    ----------
+    model : BasePhysicalModel
+        The model to draw from. This may have its own parameters which
+        will be randomly sampled with each draw.
+    num_samples : int
+        The number of samples.
+    obstable : ObsTable or List of ObsTable
+        The ObsTable(s) from which to extract information for the samples.
+    passbands : PassbandGroup or List of PassbandGroup
+        The passbands to use for generating the bandfluxes.
+    apply_obs_mask: boolean
+        If True, apply obs_mask to filter interesting indices/times.
+    time_window_offset : tuple(float, float), optional
+        A tuple specifying the time window offset (before, after) t0 in days.
+        This is used to filter the observations to only those within the specified
+        time window (t0 - before, t0 + after). If None or the model does not have a
+        t0 specified, no time window is applied.
+    obstable_save_cols : list of str, optional
+        A list of ObsTable columns to be saved as part of the results. This is used
+        to save context information about how the light curves were generated. If the column
+        is missing from one of the ObsTables, a null value such as None or NaN is used.
+        If None, no additional columns are saved.
+    param_cols : list of str, optional
+        A list of the model's parameter columns to be saved as separate columns in
+        the results (instead of just the full dictionary of parameters). These
+        must be specified as strings in the node_name.param_name format.
+        If None, no additional columns are saved.
+    output_file_path : str or Path, optional
+        The file path and name of where to save the results. If provided the results
+        are saved to this file instead of being returned directly.
+    rng : numpy.random._generator.Generator, optional
+        A given numpy random number generator to use for this computation. If not
+        provided, the function uses the node's random number generator.
+    executor : concurrent.futures.Executor, optional
```

Comment:
Let's maybe put a little more documentation about what we expect to get here? Because we are kinda supporting duck-typed executors and futures here.